model/host/host.go

```diff
@@ -3033,53 +3045,60 @@ func GetPaginatedRunningHosts(hostID, distroID, currentTaskID string, statuses [
 
 	hasFilters := false
 
-	if len(hostID) > 0 {
+	if len(opts.HostID) > 0 {
 		hasFilters = true
 
 		runningHostsPipeline = append(runningHostsPipeline, bson.M{
 			"$match": bson.M{
-				IdKey: hostID,
+				"$or": []bson.M{
+					{
+						IdKey: bson.M{"$regex": opts.HostID, "$options": "i"},
+					},
+					{
+						DNSKey: bson.M{"$regex": opts.HostID, "$options": "i"},
```

Comment:
Do we have an index on DNS name already? If not we likely need to add one (you can post this in the ops channel)

https://www.mongodb.com/docs/manual/reference/operator/query/or/#:~:text=%24or%20Clauses%20and%20Indexes&text=That%20is%2C%20for%20MongoDB%20to,will%20perform%20a%20collection%20scan.&text=MongoDB%20can%20use%20all%20but,index%20to%20support%20%24or%20clauses.